core/trino-main/src/main/java/io/trino/operator/DeduplicationExchangeClientBuffer.java

```diff
@@ -266,6 +267,12 @@ private synchronized void checkInputFinished()
                         if (failure == null) {
                             failure = taskFailure;
                         }
+                        else if (failure instanceof TrinoException && ((TrinoException) failure).getErrorCode().equals(REMOTE_TASK_FAILED.toErrorCode()) && failure != taskFailure)
+                        {
+                            // prefer to not prapagate marker REMOTE_TASK_FAILED failure externally
```

Comment:
@arhimondr I think it is not bullet proof as we may not have learned yet about actually interesting task failure when we are building final exception. Right?

Reply:
Yeah, I wonder if all of the `REMOTE_TASK_FAILED` should be ignored?